dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java

```diff
@@ -42,10 +46,58 @@
 public final class GitHubHelper {
 
     private static final Logger LOG = LoggerFactory.getLogger(GitHubHelper.class);
+    public static final String BRANCHNAME_FOR_BOT = "feature/add_dockstore_yml";
+
+    public static final String DOCKSTORE_BOT_PR_TEXT = """
+           The dockstore-bot has guessed at a .dockstore.yml for your repository.
+           
+           Please review, make appropriate changes, add any [additional keys](https://docs.dockstore.org/en/stable/assets/templates/template.html) and merge in order to complete integration with Dockstore, allowing Dockstore to keep informed of new changes to your workflow(s), notebook(s), or tool(s).
+           """;
+
+    public static final String DOCKSTORE_BOT_PR_TITLE =  "dockstore-bot inferred .dockstore.yml";
 
     private GitHubHelper() {
     }
 
+    public static URL createForkPlusPR(String inferredDockstoreYml, GitHub gitHub, String repositoryName, String targetBranch) {
+        return createForkPlusPR(inferredDockstoreYml, gitHub, repositoryName, targetBranch, BRANCHNAME_FOR_BOT, DOCKSTORE_BOT_PR_TITLE, DOCKSTORE_BOT_PR_TEXT);
+    }
+
+    /**
+     * Experimental building block, given a repository, create a fork of it (to the personal organization for the robot user) and a PR to it
+     * @param inferredDockstoreYml should be the only real change in the PR
+     * @param gitHub    The GitHub API
+     * @param repositoryName    Name of the GitHub repository (e.g. dockstore/lambda)
+     * @param targetBranch    Name of the branch in the GitHub repository to make a PR against (e.g. develop)
+     * @param branchForBot  Name of the branch in the fork
+     * @param prTitle Name of the pull request
+     * @param prDescription Description in the pull request
+     * @return TBD
+     */
+    public static URL createForkPlusPR(String inferredDockstoreYml, GitHub gitHub, String repositoryName, String targetBranch, String branchForBot, String prTitle, String prDescription) {
+        try {
+            String username = gitHub.getMyself().getLogin();
+            GHRepository targetRepository = gitHub.getRepository(repositoryName);
+            // note: this will try to turn an async call to a sync call by waiting up to half a minute, 3 seconds at a time.
+            // This did not seem consistently reliable, may need to wait more before creating a ref/branch
+            GHRepository fork = targetRepository.fork();
+            Thread.sleep(Duration.ofMinutes(1L).toMillis());
+            String masterSha = fork.getRef("heads/" + targetBranch).getObject().getSha();
+            GHRef ref = fork.createRef("refs/heads/" + BRANCHNAME_FOR_BOT, masterSha);
+            fork.createContent().content(inferredDockstoreYml).branch(ref.getRef()).message(DOCKSTORE_BOT_PR_TITLE).commit();
```

Comment:
Unfamiliar with using the kohsuke api to create a file, but should a file path be provided?